RFC0017.md

```diff
@@ -0,0 +1,150 @@
+ * the status to see if it was PMIX_MODEL_DECLARED, but it often is simpler
+ * to declare a use-specific notification callback point. In this case,
+ * we are asking to know when another programming model declared itself */
+static void release_fn(size_t evhdlr_registration_id,
+                       pmix_status_t status,
+                       const pmix_proc_t *source,
+                       pmix_info_t info[], size_t ninfo,
+                       pmix_info_t results[], size_t nresults,
+                       pmix_event_notification_cbfunc_fn_t cbfunc,
+                       void *cbdata)
+{
+    /* tell the event handler state machine that we are the last step */
+    if (NULL != cbfunc) {
+        cbfunc(PMIX_EVENT_ACTION_COMPLETE, NULL, 0, NULL, NULL, cbdata);
+    }
+    /* do whatever we want/need to do to coordinate */
+}
+
+/* event handler registration is done asynchronously because it
+ * may involve the PMIx server registering with the host RM for
+ * external events. So we provide a callback function that returns
+ * the status of the request (success or an error), plus a numerical index
+ * to the registered event. The index is used later on to deregister
+ * an event handler - if we don't explicitly deregister it, then the
+ * PMIx server will do so when it see us exit */
+static void evhandler_reg_callbk(pmix_status_t status,
+                                 size_t evhandler_ref,
+                                 void *cbdata)
+{
+    volatile int *active = (volatile int*)cbdata;
+
+    if (PMIX_SUCCESS != status) {
+        fprintf(stderr, "EVENT HANDLER REGISTRATION FAILED WITH STATUS %d, ref=%lu\n",
+                status, (unsigned long)evhandler_ref);
+    }
+    *active = status;
+}
+
+int main(int argc, char **argv)
+{
+    pmix_proc_t myproc;
+    pmix_status_t rc;
+    pmix_info_t *info;
+    char *mymodel = "MPI";
+    char *mymodelname = "FooMPI";
+    char *myversion = "1.0.0";
+    volatile int active;
+    pmix_status_t code = PMIX_MODEL_DECLARED;
+
+    /* setup to declare our programming model */
+    PMIX_INFO_CREATE(info, 4);
+    PMIX_INFO_LOAD(&info[0], PMIX_PROGRAMMING_MODEL, mymodel, PMIX_STRING);
+    PMIX_INFO_LOAD(&info[1], PMIX_MODEL_LIBRARY_NAME, mymodelname, PMIX_STRING);
+    PMIX_INFO_LOAD(&info[2], PMIX_MODEL_LIBRARY_VERSION, myversion, PMIX_STRING);
+    PMIX_INFO_LOAD(&info[3], PMIX_THREADING_MODEL, "pthread", PMIX_STRING);
+
+    /* initialize us */
+    if (PMIX_SUCCESS != (rc = PMIx_Init(&myproc, info, 4))) {
+        fprintf(stderr, "Client ns %s rank %d: PMIx_Init failed: %d\n", myproc.nspace, myproc.rank, rc);
+        exit(0);
+    }
+    PMIX_INFO_FREE(info, 4);
+    fprintf(stderr, "Client ns %s rank %d: Running\n", myproc.nspace, myproc.rank);
+
+    /* register a handler specifically for notifying us if
+     * other programming models declare themselves */
+    active = -1;
+    PMIx_Register_event_handler(&code, 1, NULL, 0,
```

Comment:
Not at the moment - however, that is being changed per the following RFC: https://github.com/pmix/RFCs/pull/18